### PR TITLE
Updated Swift docs by fixing typos and explicitly marking internal declations

### PR DESCRIPTION
**Pull request by jpsim**

### CODE REVIEW

**Comment by jpsim** [comment]
@segiddins

**Comment by alazier** [comment]
#1368 removes the private method from the Object class.

File: Realm/RealmSwift/List.swift [hunk 1]
```diff
@@ -37,7 +41,8 @@ public class ListBase: RLMListBase, Printable {
 public final class List<T: Object>: ListBase, SequenceType {
     // MARK: Properties
 
-    /// The Realm the objects in this list belong to, or `nil` if the list's owning object does not belong to a realm.
+    /// The Realm the objects in this list belong to, or `nil` if the list's owning
+    /// object does not belong to a realm (list is standalone).
```
Comment (segiddins) [inline Realm/RealmSwift/List.swift]: `the list`

File: Realm/RealmSwift/List.swift [hunk 2]
```diff
@@ -202,9 +205,9 @@ public final class List<T: Object>: ListBase, SequenceType {
 
     :param: objects A sequence of objects.
     */
-    public func append<S where S: SequenceType>(objects: S) {
-        for obj in objects {
-            _rlmArray.addObject(unsafeBitCast(obj as T, RLMObject.self))
+    public func append<S: SequenceType where S.Generator.Element == T>(objects: S) {
+        for obj in SequenceOf<T>(objects) { // Workaround for http://stackoverflow.com/questions/26918594
+             _rlmArray.addObject(unsafeBitCast(obj, RLMObject.self))
```
Comment (segiddins) [inline Realm/RealmSwift/List.swift]: that's one space too many

File: Realm/RealmSwift/List.swift [hunk 3]
```diff
@@ -247,7 +250,7 @@ public final class List<T: Object>: ListBase, SequenceType {
     }
 
     /**
-    Removes the last object in the List.
+    Removes the last object in the List. Does not remove it from the Realm.
```
Comment (segiddins) [inline Realm/RealmSwift/List.swift]: `s/it/the object/` ?

File: Realm/RealmSwift/Object.swift [hunk 4]
```diff
@@ -39,29 +38,30 @@ class Dog: Object {
...
 ### Relationships
 
-See our [Cocoa guide](http://realm.io/docs/cocoa/latest) for more details.
+See our [Cocoa guide](http://realm.io/docs/cocoa) for more details.
 */
 public class Object : RLMObjectBase, Equatable {
-    // Get the names of all properties in the object which are of type List<>
-
     /**
+    WARNING: This is a private method used by Realm that must be `public`, but is
```
Comment (segiddins) [inline Realm/RealmSwift/Object.swift]: make this a ReST `:warning:` ?

File: Realm/RealmSwift/Realm.swift [hunk 5]
```diff
@@ -197,10 +197,10 @@ public final class Realm {
 
     :param: path        Path to the file you want the data saved in.
     :param: key         64-byte key to use to encrypt the data.
-    :param: readonly    Bool indicating if this Realm is read-only (must use for read-only files).
+    :param: readOnly    Bool indicating if this Realm is read-only (must use for read-only files).
     :param: error       If an error occurs, upon return contains an `NSError` object
                         that describes the problem. If you are not interested in
-                        possible errors, pass in `NULL`.
+                        possible errors, omit the argument or pass in `nil`.
```
Comment (segiddins) [inline Realm/RealmSwift/Realm.swift]: oxford comma

File: Realm/RealmSwift/Realm.swift [hunk 6]
```diff
@@ -430,32 +430,43 @@ public final class Realm {
     public func delete(object: Object) {
         RLMDeleteObjectFromRealm(object)
     }
-
+
```
Comment (segiddins) [inline Realm/RealmSwift/Realm.swift]: trailing whitespace

File: Realm/RealmSwift/Realm.swift [hunk 7]
```diff
@@ -430,32 +430,43 @@ public final class Realm {
...
     public func delete(objects: List<Object>) {
         rlmRealm.deleteObjects(objects)
     }
+    
+    /**
+    Deletes the given objects from this Realm.
+    
+    :param: object The objects to be deleted.
+    */
+    public func delete(objects: Results<Object>) {
```
Comment (segiddins) [inline Realm/RealmSwift/Realm.swift]: pretty sure the `SequenceType` delete method covers this?

**Comment by segiddins** [comment]
@jpsim thanks so much!

File: Realm/RealmSwift/Object.swift (hunk 4 above)
Comment (jpsim) [inline Realm/RealmSwift/Object.swift]: No, we want this to be displayed to the user in jazzy docs.

File: Realm/RealmSwift/Realm.swift (hunk 7 above)
Comment (jpsim) [inline Realm/RealmSwift/Realm.swift]: The `List` and `Results` versions are optimized for those types.

**Comment by jpsim** [comment]
Merge at will, @segiddins

**Pull request merged**